File: problems/1/twosum_v1.js
/**
 * @param {number[]} array
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(array, target) {
  const length = array.length;
  for (let i = 0; i < length; i++) {
    const element = array[i];
    for (let j = i + 1; j < length; j++) {
      if (target - element == array[j]) {
        return [i, j];
      }
    }
  }
};

console.log(twoSum([2, 7, 11, 15], 9));

File: problems/1/Two_Sum_v1.cs
/// <summary>
/// Complexity Analysis
/// Time Complexity : O(n^2)
/// Space Complexity: o(1)
/// </summary>
public class Solution {
  public int[] TwoSum (int[] nums, int target) {
    int length = nums.Length;
    for (int i = 0; i < length; i++) {
      int element = nums[i];
      for (int j = i + 1; j < length; j++) {
        if (target - element == nums[j]) {
          return new int[] { i, j };
        }
      }
    }
    return new int[] { 0, 0 };
  }
}
